### ISSUE TITLE
http2: server: error reading preface from client

### ISSUE THREAD

**Issue by crypt0ginge** [gophish/gophish#1630]
When users attempt to connect the landing page  via campaigns URL they get "ERR_CONNECTION_REFUSED". Certificate is working fine (says connection is secure). Console works fine as well. Config is as follows:

{
	"admin_server": {
		"listen_url": "0.0.0.0:3333",
		"use_tls": true,
		"cert_path": "easy.crt",
		"key_path": "easy.key"
	},
	"phish_server": {
		"listen_url": "0.0.0.0",
		"use_tls": true,
		"cert_path": "easy.crt",
		"key_path": "easy.key"
	},
	"db_name": "sqlite3",
	"db_path": "gophish.db",
	"migrations_prefix": "db/db_",
	"contact_address": "test@ournetwork.com",
	"logging": {
		"filename": ""
	}
}

I have tested with TLS off and on a basic html page on port 80. It still does not load the landing page. I used the import function to clone and emulate Google Login and LastPass login.

**Comment by S0larflare** [comment]
Try putting :80 on the phish_server listen_url

Sent from my iPhone

On 25 Oct 2019, at 13:22, crypt0ginge <notifications@github.com> wrote:

﻿

When users attempt to connect the landing page via campaigns URL they get "ERR_CONNECTION_REFUSED". Certificate is working fine (says connection is secure). Console works fine as well. Config is as follows:

{
"admin_server": {
"listen_url": "0.0.0.0:3333",
"use_tls": true,
"cert_path": "easy.crt",
"key_path": "easy.key"
},
"phish_server": {
"listen_url": "0.0.0.0",
"use_tls": true,
"cert_path": "easy.crt",
"key_path": "easy.key"
},
"db_name": "sqlite3",
"db_path": "gophish.db",
"migrations_prefix": "db/db_",
"contact_address": "test@ournetwork.com<mailto:test@ournetwork.com>",
"logging": {
"filename": ""
}
}

I have tested with TLS off and on a basic html page on port 80. It still does not load the landing page. I used the import function to clone and emulate Google Login and LastPass login.
